pytests/tests/test_tdnf_python.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2019-2021 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the GNU General Public License v2 (the "License");
+# you may not use this file except in compliance with the License. The terms
+# of the License are located in the COPYING file of this distribution.
+#
+
+import os
+import sys
+import pytest
+
+
+@pytest.fixture(scope='module', autouse=True)
+def setup_test(utils):
+    global mulpkgname
+    mulpkgname = utils.config['mulversion_pkgname']
+
+    global sglpkgname
+    sglpkgname = utils.config['sglversion_pkgname']
+
+    global config
+    config = utils.config['repo_path'] + '/tdnf.conf'
+
+    path = utils.config['build_dir']
+    path += '/python/build/lib.linux-' + os.uname().machine + '-'
+    path += str(sys.version_info.major)+'.'+str(sys.version_info.minor)
+    sys.path.append(path)
+
+    global tdnf
+    import tdnf
+
+    yield
+
+    teardown_test(utils)
+
+
+def teardown_test(utils):
+    utils.run(['tdnf', 'erase', '-y', sglpkgname])
+    utils.run(['tdnf', 'erase', '-y', mulpkgname])
+    assert (utils.check_package(sglpkgname) == False)
+    assert (utils.check_package(mulpkgname) == False)
+
+
+def tdnf_py_erase(pkgs, quiet=False, refresh=False):
+    pkgs = [x for x in pkgs if x]
```

Comment:
Why check for unset entries? Can this happen?

Reply:
Yes, for some weird reason the pkg list has entries like [b' ', 'tdnf-test-one']